packages/babel-helpers/src/helpers/toSetter.ts

```diff
@@ -0,0 +1,9 @@
+/* @minVersion 7.22.0 */
+
+export default function _toSetter(fn: Function, args: any[], thisArg: any) {
+  return Object.defineProperty({}, "_", {
+    set: function (v) {
+      fn.apply(thisArg, args.concat([v]));
+    },
+  });
+}
```

Comment:
```suggestion
/* @minVersion 7.22.0 */

export default function _toSetter(fn: Function, args: any[], thisArg: any) {
  var l = args.length;
  return Object.defineProperty({}, "_", {
    set: function (v) {
      args[l] = v;
      fn.apply(thisArg, args);
    },
  });
}
```

We can make this reusable without having to re-create an array on every invocation.